### ISSUE TITLE
Optional chaining: (a?.b).c is parsed as a?.b.c

### ISSUE THREAD

**Issue by thorn0** [typescript-eslint/typescript-eslint#1139]
https://github.com/prettier/prettier/pull/6657 is blocked by this issue.

**What code were you trying to parse?**

```ts
(a?.b).c
```

**What did you expect to happen?**

The outer node should be of type `MemberExpresion`, not `OptionalMemberExpression`.

**What actually happened?**

Both member expressions are `OptionalMemberExpression` as if the parens weren't there (`a?.b.c`).

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `2.5.1-alpha.3` |
| `TypeScript`                           | `3.7.0-rc` |
| `node`                                 | `10.15.0` |
| `npm`                                  | `6.11.3` |

**Comment by bradzacher** [comment]
I don't understand why this case is handled differently. It seems to me like this is semantically and syntactically no different.

Unless I'm mistaken, the parenthesis don't actually change the output or order of execution?

**Comment by thorn0** [comment]
TS compiles

- `a?.b.c` to: `(_a = a) === null || _a === void 0 ? void 0 : _a.b.c`
- `(a?.b).c` to: `((_b = a) === null || _b === void 0 ? void 0 : _b.b).c`

[playground](https://www.typescriptlang.org/play/?ts=3.7-Beta&ssl=5&ssc=9&pln=5&pc=17#code/PTAEAEBcGcFoEsDmA7A9gJwKYFgBQA3AQ3VAA9QBeUQgfgDoAjOgYwG488QIYEUMcCxUAE9KoABS1GAShasgA)

**Comment by bradzacher** [comment]
Hmmm yes, I am mistaken - it does change the order of execution and thus the return values and types.

```ts
type T = Record<string, Record<string, string>>;
declare const x: T | undefined;

const v1 = x?.a.b;
const v2 = (x?.a).b;
//         ^^^^^^ Object is possibly 'undefined'.
const v2 = (x?.a)?.b;
```

**Comment by thorn0** [comment]
Yep, it's tricky. And `x?.a?.b` and `(x?.a)?.b` are semantically equivalent, right?
